_threat-model-log4net.adoc

```diff
@@ -0,0 +1,76 @@
+////
+    Licensed to the Apache Software Foundation (ASF) under one or more
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+         https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+////
+
+[#threat-log4net]
+=== Log4Net threat model
+
+Below we share the threat model specific to link:/log4net[log4net].
+
+[#threat-log4net-code-signing]
+==== Code signing
+
+All Logging Services software release distributions are signed using GPG using a key from the Logging Services PMC https://downloads.apache.org/logging/KEYS[KEYS file].
+Information on how to verify releases signatures are explained further in xref:download.adoc[the Download page].
+Thus, GPG signatures should be validated in your build process.
+
+[#threat-log4net-config-sources]
+==== Configuration sources
+All configuration sources to an application must be trusted by the programmer.
+When loading a configuration file from disk (especially when a monitor interval is configured to reload the file periodically), the location of the configuration file must be kept safe from unauthorized modifications.
+Similarly, when loading a configuration file over the network such as through HTTP, this should be configured to use TLS or a secure connection in general with strong authentication guarantees.
+This remote location must be kept safe from unauthorized modifications.
+
+[#threat-log4net-parametrized-logging]
+==== Parameterized logging
+
+When using a log message containing template parameters like `{0}`, only the format string is evaluated for parameters to be substituted.
+The message parameters themselves are not evaluated for parameters; they are only included in the format string corresponding to their template position.
+The conversion of message parameters into a string is done on-demand depending on the layout being used.
+When structure-preserving transformations of log message data are required, the `Message` API should be used for logging structured data combined with a structured layout (e.g., `XmlLayout`).
```

Comment:
Does Log4Net have a `Message` API? (`Log masking` section below also has a mention of it.)

Reply:
Yes, as mentioned in PMC channel, you can call
`logger.Warn(object? message)`
Log4Net calls .ToString() for the "message" to serialize it.

Reply:
I don't see a dedicated type/contract, but a generic `object` (contrary to Log4j, which uses the `Message` interface, hence the wording), am I mistaken?

Log4j's `Message` type provides contract(s) on how shall its content be encoded for a certain formatting; JSON, XML, etc. AFAIU, Log4Net does not have such a type and calling `.ToString()` is not structure-preserving. I'd reword this sentence as follows:

```suggestion
When structure-preserving transformations of log data are required, a structured layout (e.g., `XmlLayout`) should be used.
```